### PR TITLE
Monorepo

### PR DESCRIPTION
**Pull request by HenriBeck**
What is already done:
- Moved whole jss package into packages/jss
- Initialized lerna
- Added universal build scripts
- Karma runs the test succesfully

Still left to do:
- There is a problem with the snapshoting

### CODE REVIEW

**Comment by HenriBeck** [comment]
Fixed snapshot testing

**Comment by kof** [comment]
@HenriBeck we have this tests/index.html, which is using the tmp/tests.js. I think we can get rid of it. The use case was to run tests from a static page, but we can also run it through the local tests server once it is running. It was easy to just open a html file without any server on any browser.

**Comment by kof** [comment]
@HenriBeck let me know when to try this out

**Comment by HenriBeck** [comment]
@kof should be ready to try out

**Comment by HenriBeck** [comment]
The problem is when we merge any pr, we create conflicts because literally every file has changed. So if we could move forward before any other PRs are merged it would be great.

**Comment by kof** [comment]
oh yeah, sorry for that, we need to merge it asap, I will look into it today

**Comment by kof** [comment]
are you available on some chat? in case I have things to discuss

**Comment by HenriBeck** [comment]
you can just email me at henribeck.dev@gmail.com

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,4 @@ lib
 dist
 .env
 coverage
+packages/**/lincense.md
```
Comment (kof) [inline .gitignore]: do we need a license file in each package?

File: docs/abstractions.md [hunk 2]
```diff
@@ -4,4 +4,4 @@ JSS core library is a rendering engine. We aim to make it small, fast and unopin
 It is often too low level for applications.
 That's why there is a bunch of higher level libraries on top of it, which provide more opinionated sugar APIs and framework integrations.
 
-[See all projects](./projects.md)
```
Comment (kof) [inline docs/abstractions.md]: driveby commits like this making PR larger and harder to merge

File: scripts/build-lib.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+/* eslint-disable no-console */
```
[the remark this answers is not in the document]
Comment (kof) [reply scripts/build-lib.js]: is it to run those scripts over each package? doesn't lerna has something to do that in a more concise way? any other options?

[the remark this answers is not in the document]
Comment (HenriBeck) [reply scripts/build-lib.js]: No, we would need to install babel-cli for every package, there is no way to use those commands inside npm scripts without installing it inside every package.

That way we only need one script and not have multiple build scripts for every package.

File: .gitignore (hunk 1 above)
Comment (HenriBeck) [reply .gitignore]: It just copies it when the build script runs from the root path. I think every npm package should have a LICENSE file, I don't think we lose anything without it.

File: scripts/build-lib.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (kof) [reply scripts/build-lib.js]: what about ../../node_modules/.bin/{script}

[the remark this answers is not in the document]
Comment (HenriBeck) [reply scripts/build-lib.js]: Still, we would need a `build:lib`, `build:dist` and maybe `build:flow` script in every package.
I think this one is more maintainable. This approach is also used by storybook.

**Comment by kof** [comment]
Ok, found again time, working on it.

**Comment by kof** [comment]
there are so many questions open that I think we can't make it in one PR and we need to start merging it and iterate

**Comment by kof** [comment]
Do we want to have a single version number for all plugins or separate?

**Comment by kof** [comment]
merged 🎉

**Comment by HenriBeck** [comment]
I think we should have one single version. One single version for the core and the plugins which should make it clear which version you need to install when installing a plugin.

**Comment by HenriBeck** [comment]
I would start importing the plugins then next. Should find some time on the weekend for that.

**Comment by kof** [comment]
yeah, me too, even though react-jss could be a separate version, I think its easier that way

**Comment by kof** [comment]
On the related note, created

https://github.com/cssinjs/jss/issues/745 

https://github.com/cssinjs/jss/issues/744
